src/frontend/src/eth/utils/qr-code.utils.ts

```diff
@@ -19,13 +23,58 @@ export const decodeQrCode = ({
 		return { status: 'cancelled' };
 	}
 
-	const payment = decodePayment(code);
+	const payment = decodeQrCodeUrn(code);
 
 	if (isNullish(payment)) {
 		return { status: 'success', destination: code };
 	}
 
-	const { token, identifier: destination, amount } = payment;
+	const { prefix, destination, amount, address, ethereumChainId, functionName } = payment;
+
+	const normalizeChainId = (chainId: string): string => {
+		if (chainId.startsWith('0x')) {
+			return parseInt(chainId, 16).toString();
+		}
+		return chainId;
+	};
+
+	const matchEthereumToken = (
+		address: string | undefined,
+		functionName: string | undefined,
+		ethereumChainId: string | undefined
+	): string | undefined => {
+		const parsedEthereumChainId = nonNullish(ethereumChainId)
+			? normalizeChainId(ethereumChainId)
+			: 1n;
```

Comment:
wouldn't be safer to not fallack on what I guess is ethereum production and ignore or throw an error if the chain is not supported?

Reply:
Yes, it makes sense

Reply:
Generally speaking I would avoid starting adding deep comparison has the root feature of Oisy is and should remain to be the usage of JavaScript symbol - we are discussing a special case here.

Ok, works for me, if it can be omitted, then we do not crash an error but fallback. However, then it should not be `1n` but the chain id of the currently selected network I think - i.e. either Ethereum or Sepolia.

Reply:
perfect! changing it

Reply:
perfect! changing it